### PR TITLE
2024-Intake

### PR DESCRIPTION
**Pull request by theAbishek718**
Adds Intake subsystem with pre-assigned speeds that can be changed when we test. It also does not include commands in RobotContainer.

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -160,6 +160,7 @@ private void configureButtonBindings() {
                                 drive.getPose().getTranslation(),
                                 AllianceFlipUtil.apply(new Rotation2d()))))
                 .ignoringDisable(true));
+    // Mapped buttons on controller to intake, release, or stop.
```
Comment (EQMoore) [inline src/main/java/frc/robot/RobotContainer.java:163]: Remove line 163

File: src/main/java/frc/robot/Schematic.java [hunk 2]
```diff
@@ -58,6 +62,8 @@ public class Schematic {
         FRONT_RIGHT_STEERING_ID = 8;
         FRONT_RIGHT_CANCODER_ID = 16;
         GYRO_ID = 0;
+
+        INTAKE_ID = 12;
       }
       default -> {
         POWER_DIST_ID = 20;
```
Comment (EQMoore) [inline src/main/java/frc/robot/Schematic.java:69]: Should be zero

File: src/main/java/frc/robot/subsystems/IntakeNote/IntakeNote.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+public class IntakeNote extends SubsystemBase {
+  /** Creates a new Intake. */
+  private final IntakeNoteIO intakeNoteIO;
+
+  private final IntakeNoteIOInputsAutoLogged intakeInputs;
+  // TODO: Change these speeds as needed once we test.
+  private final double intakeSpeed = 0.2;
+  private final double releaseSpeed = -0.2;
+  private final double stop = 0.0;
+
```
Comment (EQMoore) [inline src/main/java/frc/robot/subsystems/IntakeNote/IntakeNote.java:21]: lines 18-20 should be constants in the IntakeConstants file

**Review by EQMoore — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/IntakeNote/IntakeNote.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import org.littletonrobotics.junction.Logger;
+
+public class IntakeNote extends SubsystemBase {
+  /** Creates a new Intake. */
+  private final IntakeNoteIO intakeNoteIO;
+
+  private final IntakeNoteIOInputsAutoLogged intakeInputs;
+  // TODO: Change these speeds as needed once we test.
```
Comment (admlvntv) [inline src/main/java/frc/robot/subsystems/IntakeNote/IntakeNote.java:17]: is this resolved?

File: src/main/java/frc/robot/subsystems/IntakeNote/IntakeNote.java
Comment (admlvntv) [inline src/main/java/frc/robot/subsystems/IntakeNote/IntakeNote.java:1]: remove some of the comments from this file, maybe replace with javadoc

File: src/main/java/frc/robot/subsystems/IntakeNote/IntakeNoteIO.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import org.littletonrobotics.junction.AutoLog;
+
+public interface IntakeNoteIO {
+  @AutoLog
+  class IntakeNoteIOInputs {
+    // I/p for intake
+    public boolean seesNote = false; // Need to communicate with Vision
+    public boolean armSwitchClosed = false;
+
+    // O/p for intake
```
Comment (admlvntv) [inline src/main/java/frc/robot/subsystems/IntakeNote/IntakeNoteIO.java:12]: ???

File: src/main/java/frc/robot/subsystems/IntakeNote/IntakeNoteIOPhysical.java
Comment (admlvntv) [inline src/main/java/frc/robot/subsystems/IntakeNote/IntakeNoteIOPhysical.java:1]: remove some of the comments from this file, maybe replace with javadoc

**Review by admlvntv — CHANGES_REQUESTED** [review]
some changes

File: src/main/java/frc/robot/subsystems/IntakeNote/IntakeNoteIO.java (hunk 5 above)
Comment (theAbishek718) [reply src/main/java/frc/robot/subsystems/IntakeNote/IntakeNoteIO.java:12]: These were from earlier in build season and were just serving as a thought. I will get rid of them, thank you for telling me.

File: src/main/java/frc/robot/subsystems/IntakeNote/IntakeNote.java (hunk 4 above)
Comment (theAbishek718) [reply src/main/java/frc/robot/subsystems/IntakeNote/IntakeNote.java:17]: No, we never got a prototype to test with cause it took too long and they just went to comp bot cad. So we will have to test speeds when we get the robot. But it shouldn't take too long.

**Review by EQMoore — APPROVED** [review]
Lgtm

File: src/main/java/frc/robot/subsystems/IntakeNote/IntakeNote.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
```
Comment (admlvntv) [inline src/main/java/frc/robot/subsystems/IntakeNote/IntakeNote.java:3]: remove header

File: src/main/java/frc/robot/subsystems/IntakeNote/IntakeNote.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+          Logger.recordOutput("IntakeNote/DesiredSpeed", IntakeConstants.INTAKE_SPEED);
+          intakeNoteIO.setSpeed(IntakeConstants.INTAKE_SPEED);
+        },
+        this);
+  }
+
+  public Command release() {
+    return Commands.run(
+        () -> {
+          Logger.recordOutput("IntakeNote/DesiredReleaseSpeed", IntakeConstants.RELEASE_SPEED);
```
Comment (admlvntv) [inline src/main/java/frc/robot/subsystems/IntakeNote/IntakeNote.java:42]: Make this use DesiredSpeed

File: src/main/java/frc/robot/subsystems/IntakeNote/IntakeNote.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+          intakeNoteIO.setSpeed(IntakeConstants.RELEASE_SPEED);
+        },
+        this);
+  }
+
+  public Command stop() {
+    return Commands.run(
+        () -> {
+          intakeNoteIO.setSpeed(IntakeConstants.STOP_SPEED);
```
Comment (admlvntv) [inline src/main/java/frc/robot/subsystems/IntakeNote/IntakeNote.java:51]: update the DesiredSpeed

**Review by admlvntv — CHANGES_REQUESTED** [review]
almost there

**Review by admlvntv — APPROVED** [review]
lgtm

**Pull request merged**